src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -282,29 +282,26 @@ struct
           account_hash
       in
       List.iter addresses_and_hashes ~f:(fun (addr, hash) ->
-          set_hash t addr hash )
+          self_set_hash t addr hash )
 
     (* if the mask's parent sets an account, we can prune an entry in the mask
        if the account in the parent is the same in the mask *)
     let parent_set_notify t account =
       assert_is_attached t ;
-      match find_location t (Account.public_key account) with
+      match self_find_location t (Account.public_key account) with
       | None -> ()
       | Some location -> (
-        match find_account t location with
+        match self_find_account t location with
         | Some existing_account ->
-            if
-              Key.equal
-                (Account.public_key account)
-                (Account.public_key existing_account)
-            then remove_account_and_update_hashes t location
+            if Account.equal account existing_account then
```

Comment:
Do all fields have to match?  Is it enough that the key and balance are the same? I don't understand the receipt chain hash, so I don't know, myself.

Reply:
Yeah, otherwise we're losing information.